### PR TITLE
Ad/Sumcheck Tests

### PR DESCRIPTION
**Pull request by arijitdutta67**
# Description

This is a small PR doing the followings
- Adds a test file for `barycentric_data.hpp`
-  Modifies and expands on the tests on `multivariates.hpp`
- Adds some explanatory comments and removes some typos on various files of the `sumcheck` module. 

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [ ] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [ ] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+using namespace honk::sumcheck;
+namespace test_sumcheck_polynomials {
+
+template <class FF> class BarycentricDataTests : public testing::Test {};
+
+using FieldTypes = testing::Types<barretenberg::fr>;
+TYPED_TEST_SUITE(BarycentricDataTests, FieldTypes);
+
+#define BARYCENTIC_DATA_TESTS_TYPE_ALIASES using FF = TypeParam;
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp:15]: There are tests along these lines in univariate.test.cpp also. Do you mind unifying the tests, either moving some of those here, or moving these tests into that file?

File: cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+TYPED_TEST(BarycentricDataTests, Extend)
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp:17]: There are tests along these lines in univariate.test.cpp also. Do you mind unifying the tests, either moving some of those here, or moving these tests into that file?

File: cpp/src/aztec/honk/sumcheck/polynomials/multivariates.hpp [hunk 3]
```diff
@@ -125,11 +125,11 @@ template <class FF_, size_t num_polys> class Multivariates {
      * groups    vertex terms              collected vertex terms               groups after folding
      *     g0 -- v0 (1-X1)(1-X2)(1-X3) --- (v0(1-X3) + v1 X3) (1-X1)(1-X2) ---- (v0(1-u3) + v1 u3) (1-X1)(1-X2)
      *        \- v1 (1-X1)(1-X2)  X3   --/                                  --- (v2(1-u3) + v3 u3) (1-X1)  X2
-     *     g1 -- v2 (1-X1)  X2  (1-X3) --- (v1(1-X3) + v2 X3) (1-X1)  X2  -/ -- (v4(1-u3) + v5 u3)   X1  (1-X2)
+     *     g1 -- v2 (1-X1)  X2  (1-X3) --- (v2(1-X3) + v3 X3) (1-X1)  X2  -/ -- (v4(1-u3) + v5 u3)   X1  (1-X2)
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/multivariates.hpp:125]: Thanks for noticing this.

File: cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp [hunk 4]
```diff
@@ -35,15 +33,19 @@ TYPED_TEST(MultivariatesTests, Constructor)
...
-    (v01 * (1-X1) + v11 * X1) *   X2   ~~>    (v00 * (1-u2) + v01 * u2) * (1-X1)
-  + (v00 * (1-X1) + v10 * X1) * (1-X2) ~~>                                    + (v11 * u2 + v10 * (1-u2)) * X1
+                                       After folding
+           v10 ------ v11                  ~~>
+            |          |                   ~~>
+    X1(msb) |          |                   ~~>
+            | X2(lsb)  |                   ~~>
+           v00 ------ v01
+f0(X1,X2) =
+    v00 * (1-X1)(1-X2) + v01 * (1-X1) *X2
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp]: Could you clean this up a bit? It's kind of messy.

File: cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp [hunk 5]
```diff
@@ -110,8 +112,194 @@ TYPED_TEST(MultivariatesTests, FoldTwoRoundsGeneric)
 
     FF round_challenge_1 = FF::random_element();
     FF expected_val = expected_lo * (FF(1) - round_challenge_1) + expected_hi * round_challenge_1;
-
     multivariates.fold(multivariates.folded_polynomials, multivariate_n >> 1, round_challenge_1);
     EXPECT_EQ(multivariates.folded_polynomials[0][0], expected_val);
 }
+
+// Three rounds
+/*f0(X1, X2, X3) = v000(1-X1)(1-X2)(1-X3) + v001(1-X1)(1-X2)X3 + v010(1-X1)X2(1-X3) + v011(1-X1)X2X3
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp]: Looks correct. Could you also align this a bit more clearly so it's easier to read?

**Review by codygunton — CHANGES_REQUESTED** [review]
Thanks for writing these. I make only small requests for changes.

File: cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp (hunk 2 above)
Comment (arijitdutta67) [reply cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp:17]: Thanks for noticing that. I have moved the tests `BarycentricData2to3` and `BarycentricData5to6` from `univariate.test.cpp` to `barycentric_data.test.cpp`.

File: cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp (hunk 4 above)
Comment (arijitdutta67) [reply cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp]: Done.

File: cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp (hunk 5 above)
Comment (arijitdutta67) [reply cpp/src/aztec/honk/sumcheck/polynomials/multivariates.test.cpp]: Done.

**Pull request merged**